### ISSUE TITLE
Document minimum supported Gradle version

### ISSUE THREAD

**Issue by AppearamidGuy** [dependency-check/dependency-check-gradle#348]
tl;dr: based on some basic testing, plugin works with Gradle 7.6+ and with 5.0-6.5.1. Also see the last paragraph

I started investigating this to better understand how difficult it would be to support configuration cache. However it might be also useful in the context of https://github.com/jeremylong/DependencyCheck/issues/5791. 

All test were performed with plugin version 8.3.1 and a basic java project.

The plugin is actually very backwards compatible and the earliest version of Gradle where the plugin applies successfully is 3.3 (released Jan 2017!). But the first version that can run tasks is 5.0 as it's the first version with Groovy >= 2.5.0. Earlier plugin versions (8.0.1 and below) actually can run with Gradle 3.3+, but 8.0.2+ can't. Finally, plugin can't be applied in Gradle versions 6.6-7.5.1 because of java version in jackson.

Declaring 7.6 as minimum supported Gradle version can be reasonable considering that 7.x is in maintenance only mode and it's the first version where plugin actually works. It is definitely enough to support configuration caching and also has some nice features like ["Using dependency resolution results as task inputs"](https://docs.gradle.org/7.5/userguide/more_about_tasks.html#sec:task_input_using_dependency_resolution_results) which should help with #100
